assignments/prototypes.js

```diff
@@ -41,8 +41,55 @@
 
 // Test you work by un-commenting these 3 objects and the list of console logs below:
 
-/*
-  const mage = new Humanoid({
+function GameObject(gameObjectItem) {
+  this.createdAt = gameObjectItem.createdAt;
+  this.dimensions = gameObjectItem.dimensions;
+};
+
+GameObject.prototype.destroy = function () {
+  return `Object was removed from the game.`
+};
+
+function CharacterStats(statsItem) {
+  GameObject.call(this, statsItem);
+  this.healthPoints = statsItem.healthPoints;
+  this.name = statsItem.name;
+};
+
+CharacterStats.prototype.takeDamage = function () {
+  return `${this.name} took damage.`
+};
+
+CharacterStats.prototype = Object.create(GameObject.prototype);
```

Comment:
This line was tripping a lot of people up. This line tells the CharacterStats prototype to _extend_ the GameObject prototype, which gives it all of the methods associated with GameObject. The trouble is, it erased your takeDamage function, because that's not part of the GameObject prototype. You have to call this _first_, and then add any additional methods you want on the CharacterStats prototype _after_.

```suggestion

```